integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metrics_number_dimension
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+	"strings"
+	"testing"
+	"time"
+)
+
+const configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const configJSON = "/config.json"
+const namespace = "MetricNumberDimensionTest"
+const instanceId = "InstanceId"
+const append = "append"
+
+//Let the agent run for 2 minutes. This will give agent enough time to call server
+const agentRuntime = 2 * time.Minute
+
+var targetString = "max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number"
```

Comment:
This feels like bad code smell to me. It took me a bit to realize that this global gets modified in the tests. I think that we should not modify this inside the test. We should create a new string off of this template, and then pass _that_ value into the `outputLogContainsTarget` function.